dts/bindings/clock/raspberrypi,pico-xosc.yaml

```diff
@@ -11,4 +11,6 @@ include: raspberrypi,pico-clock.yaml
 properties:
   startup-delay-multiplier:
     type: int
-    description: Startup delay multiplier
+    default: 1
```

Comment:
Since this is set to 1 here, I would suggest removing https://github.com/zephyrproject-rtos/zephyr/blob/0954e443f38775a391aa5fd4bca89159ae6648f9/dts/arm/raspberrypi/rpi_pico/rp2040.dtsi#L165 so the default is consistent across RP2040 and RP2350.

Reply:
@ajf58 can you please address this comment.
Either by following the suggestion or reply why that's not a good idea in this case.

Reply:
> Either by following the suggestion or reply why that's not a good idea in this case.

@xudongzheng was the original author of this change, so they're kinda arguing with their past self here, with me as a proxy. So the full suite of options are:

1. Leave as I've done. This will not change the behaviour of the in-tree RP2040-based boards. This, implicitly, has been approved many many times over as part of another PR.
2. Only remove `startup-delay-multiplier` from rp2040.dtsi. This will effectively revert what @xudongzheng added in the first place.
3. Remove `startup-delay-multiplier` from rp2040.dtsi and only add it to the in-tree boards where it is known to be needed, i.e. cross match against https://github.com/raspberrypi/pico-sdk/tree/master/src/boards/include/boards . This, IMHO is arguably what the original PR that introduced the delay should have done: it fixes things that are known to need this, and doesn't alter out-of-tree boards' behaviour.

Reply:
> @xudongzheng was the original author of this change, so they're kinda arguing with their past self here,

Ever written code where you later discover this should probably have been done differently or at another location ?
I have done that many times.

Of course I cannot answer for @xudongzheng , but to me this looks like such a case, and when this PR broadly states `improvements`, then this seems to be one such improvement which can be taken in.

Though I agree it more seems like something which should be a regular comment, and not a change request.

> Remove startup-delay-multiplier from rp2040.dtsi and only add it to the in-tree boards where it is known to be needed, i.e. cross match against https://github.com/raspberrypi/pico-sdk/tree/master/src/boards/include/boards . This, IMHO is arguably what the original PR that introduced the delay should have done: it fixes things that are known to need this, and doesn't alter out-of-tree boards' behaviour.

To me this also sounds like the ideal approach.

And as PR author you are free to decide if you want to include this change or not.

But thanks for replying to the comment, as that is the only way for others to know if it has been seen / considered.

Reply:
> you are free to decide if you want to include this change or not.

I'm not going to be extending the scope of this change beyond what it currently does, i.e. "prevent out-of-tree boards based that use this DTS binding requiring to define a parameter." 

It's a remarkable coincidence that in the "future" (subsequent commits)[1] it means that if another SoC uses this same driver it won't have to keep defining something to be equal to one. Someone should make one of those and put it on some boards... maybe get a PR to put them in-tree.

The value is board, not SoC specific, and really shouldn't be in rp2040.dtsi whatsoever. @xudongzheng , perhaps you can raise an issue referencing back to your own PR with these suggestions? I don't have time to do that, nor to action it.

[1] or ~weeks in the past for those humans who measure their life like this.

Reply:
Adding a default value on the yaml side will not change the value explicitly specified in dts.
In other words, the current behavior will not change even if this change is.
Therefore, I think it would be better to extract this into a separate PR.

As @xudongzheng says, deleting the value on the dtsi side would be better.
Can you please address this after this PR is completed? > @xudongzheng

Reply:
I have it in my internal RP2350 notes to follow up on `startup-delay-multiplier` with a PR if it's not resolved here.

The important part of #78200 is making the delay configurable, not so much what the default value is. I prefer 64, @soburi somewhere in #77368 mentioned preferring 1 to be consistent with the Pico SDK. I'm inclined to defer to his opinion as the maintainer.

Reply:
@xudongzheng 
The dts setting takes precedence over the default, so even if we do not fix it in this PR, the behavior will not change. Let's deal with it in another PR.